src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -824,7 +824,10 @@ self =>
 
       def mkSelection(t: Tree) = {
         def sel = atPos(opPos union t.pos)(Select(stripParens(t), op.encode))
-        if (targs.isEmpty) sel else atPos(left.pos)(TypeApply(sel, targs))
+        if (targs.isEmpty) sel
+        else atPos(sel.pos union targs.last.pos withPoint sel.pos.point) {
```

Comment:
`sel` is a `def` (for `def`ensive tree duplication?) I suggest turning it into a `val`, along with its position, so that both are evaluated once

Reply:
The follow-up was the second commit, this diff was outdated.